bridge/src/main/java/com/livefront/bridge/wrapper/WrapperUtils.java

```diff
@@ -0,0 +1,39 @@
+package com.livefront.bridge.wrapper;
+
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.os.Parcel;
+
+import java.util.Set;
+
+/**
+ * Handles the wrapping and unwrapping of certain {@link android.os.Parcelable} objects that use
+ * native code to optimize their {@code Parcelable} implementations. When placed into a {@link
+ * Bundle} unwrapped, these objects will cause a crash if the {@code Bundle} is written to a
+ * {@link android.os.Parcel} that then calls {@link Parcel#marshall()}.
+ */
+public class WrapperUtils {
+
+    public static void unwrapOptimizedObjects(Bundle bundle) {
+        Set<String> keys = bundle.keySet();
+        for (String key : keys) {
+            if (bundle.get(key) instanceof BitmapWrapper) {
```

Comment:
Is your plan just to have a few `if` statements here for each new type you need to wrap?  My only thought would be to create a `Wrapper` interface that each wrapper implements so it could handle it on its own:

```
if (bundle.get(key) instanceof Wrapper) {
    Wrapper wrapper = (Wrapper) bundle.get(key);
    bundle.putParcelable(key, wrapper.unwrap());
}
```

However, if you only have a couple of these, that's probably not necessary, and you'd lose the symmetry you have down in `wrapOptimizedObjects`, because I don't think something like that would work there.